File: example-codes/stt-openai.js
const OpenAI = require('openai');
const fs = require('fs');

if (!globalThis.File) {
    globalThis.File = require('node:buffer').File;
}

async function main() {
    const client = new OpenAI({
        baseURL: 'https://stt-api.cantonese.ai/v1/',
        apiKey: 'XXXXXXX'
    });

    const audioFile = fs.createReadStream('audio.wav');
    const transcript = await client.audio.transcriptions.create({
        file: audioFile,
        language: 'zh'
    });
    console.log(transcript.text);
}

main().catch(console.error);

File: example-codes/stt-openai.py
from openai import OpenAI

client = OpenAI(base_url="https://stt-api.cantonese.ai/v1/", api_key="XXXXXXX")

audio_file = open("audio.wav", "rb")
transcript = client.audio.transcriptions.create(file=audio_file, model="", language="zh")
print(transcript.text)